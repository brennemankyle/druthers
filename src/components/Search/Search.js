import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'
import AppPropTypes from '../../utils/AppPropTypes'

let Search = (props, ref) => {
  return <input
    className={props.className + ' search'}
    ref={ref}
    disabled={props.styles_disabled}
    type="text"
    placeholder={props.placeholder}
    onChange={props.onChange}
    onKeyDown={props.onKeyDown}
    onFocus={props.onFocus}
    onBlur={props.onBlur}
    value={props.searchText} />
}

Search = forwardRef(Search)

Search.propTypes = {
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  searchText: PropTypes.string,
  onKeyDown: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  onBlur: PropTypes.func.isRequired,
  hide: PropTypes.bool.isRequired,
  ...AppPropTypes.styles,
}

export default Search
