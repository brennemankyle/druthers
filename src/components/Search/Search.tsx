import React, { forwardRef, MouseEventHandler, ReactElement } from "react";

interface Props {
  placeholder: string;
  onChange: MouseEventHandler<HTMLInputElement>;
  searchText: string;
  onKeyDown: MouseEventHandler<HTMLInputElement>;
  onFocus: MouseEventHandler<HTMLInputElement>;
  onBlur: MouseEventHandler<HTMLInputElement>;
  hide: boolean;
}

const Search = forwardRef(function Search(props: Props, ref): ReactElement {
  return (
    <input
      className={props.className + " search"}
      ref={ref}
      disabled={props.styles_disabled}
      type="text"
      placeholder={props.placeholder}
      size={props.styles_search_size}
      onChange={props.onChange}
      onKeyDown={props.onKeyDown}
      onFocus={props.onFocus}
      onBlur={props.onBlur}
      value={props.searchText}
    />
  );
});

export default Search;
