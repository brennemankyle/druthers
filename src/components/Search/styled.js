import styled from "@emotion/styled";
import RawSearch from "./Search";

const Search = styled(RawSearch)`
  border: none;
  outline: 0;
  font-size: 1em;
  font-family: inherit;
  background-color: transparent;
  &::-ms-clear {
    display: none;
  }
  margin-top: calc(${props => props.styles_selection_margin} / 2);
  margin-bottom: calc(${props => props.styles_selection_margin} / 2);
  padding-top: ${props => props.styles_selection_paddingTop};
  padding-bottom: ${props => props.styles_selection_paddingBottom};
  padding-right: ${props => props.styles_selection_paddingRight};
  padding-left: ${props => props.styles_selection_paddingLeft};
  ${props =>
    props.styles_rightToLeft &&
    `transform: scaleX(-1);
    text-align: right;`}
  ${props =>
    props.hide &&
    `opacity: 0;
    width: 0;
    padding: 0;
    margin: 0;`}
  ${props => props.styles_disabled && `cursor: not-allowed;`}
`;

export { Search };
