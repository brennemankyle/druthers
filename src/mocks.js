let mockStyles = {
  colors: {}
}

let mockItem = {
  value: '42',
  label: 'Meaning of life',
}

let mockItemList = [
  {value: '1', label: 'Item 1'},
  {value: '2', label: 'Item 2'},
  {value: '3', label: 'Item 3'},
  {value: '4', label: 'Item 4'},
  {value: '5', label: 'Item 5'},
]

export { mockStyles, mockItem, mockItemList }
