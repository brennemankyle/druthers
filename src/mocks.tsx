import React from "react";
import defaultProps from "./utils/defaultProps";
import { withKeys, noop } from "./utils/utils";

let mockStyles = {
  styles_width: 600,
  styles_multiple: false,
  styles_disabled: false,
  styles_hasSelection: false,
  styles_hasOptions: true,
  styles_optionHighlighted: [],
  styles_rightToLeft: false,
  ...withKeys(defaultProps, "styles_"),
};

let mockItem = {
  value: "42",
  label: "Meaning of life",
};

let mockItemList = [
  { value: "1", label: "Item 1" },
  { value: "2", label: "Item 2" },
  { value: "3", label: "Item 3" },
  { value: "4", label: "Item 4" },
  { value: "5", label: "Item 5" },
];

let mockEvent = {
  preventDefault: noop,
};

let MockElement = (props: any) => <div data-props={props} />;
let MockInput = (props: any) => <input type="text" data-props={props} />;
let MockSvg = () => (
  <svg>
    <path d="M0 0L5 5Z" />
  </svg>
);

let mockRect = {
  top: 10,
  bottom: 110,
  left: 20,
  right: 220,
  width: 200,
  height: 100,
  x: 15,
  y: 25,
};

export {
  mockStyles,
  mockItem,
  mockItemList,
  mockEvent,
  MockElement,
  MockInput,
  MockSvg,
  mockRect,
};
