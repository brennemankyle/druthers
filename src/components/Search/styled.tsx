import styled from "@emotion/styled";
import RawSearch from "./Search";

const Search = styled(RawSearch)(
  props => `
  border: none;
  outline: 0;
  font-size: 1em;
  font-family: inherit;
  background-color: transparent;
  &::-ms-clear {
    display: none;
  }
  margin-top: calc(${props.styles_selection_margin} / 2);
  margin-bottom: calc(${props.styles_selection_margin} / 2);
  padding-top: ${props.styles_selection_paddingTop};
  padding-bottom: ${props.styles_selection_paddingBottom};
  padding-right: ${props.styles_selection_paddingRight};
  padding-left: ${props.styles_selection_paddingLeft};
  ${
    props.styles_rightToLeft
      ? `transform: scaleX(-1);
        text-align: right;`
      : ``
  }
  ${
    props.hide
      ? `opacity: 0;
        width: 0;
        padding: 0;
        margin: 0;`
      : ``
  }
  ${props.styles_disabled ? `cursor: not-allowed;` : ``}
`
);

export { Search };
