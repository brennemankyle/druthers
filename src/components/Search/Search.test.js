import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { mockStyles } from '../../mocks'
import Search from './Search'

const placeholder = 'Placeholder'
const onChange = jest.fn()
const onFocus = jest.fn()
const onBlur = jest.fn()
const onKeyDown = jest.fn()

it('renders', () => {
  const wrapper = shallow(<Search
    {...mockStyles}
    placeholder={placeholder}
    onChange={onChange}
    onKeyDown={onKeyDown}
    onFocus={onFocus}
    onBlur={onBlur}
    hide={false}
    className="test" />)

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders disabled', () => {
  let styles = {
    ...mockStyles,
    styles_disabled: true,
  }

  const wrapper = shallow(<Search
    {...styles}
    placeholder={placeholder}
    onChange={onChange}
    onKeyDown={onKeyDown}
    onFocus={onFocus}
    onBlur={onBlur}
    hide={false}
    className="test" />)

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders searchText', () => {
  const wrapper = shallow(<Search
    {...mockStyles}
    placeholder={placeholder}
    onChange={onChange}
    onKeyDown={onKeyDown}
    onFocus={onFocus}
    onBlur={onBlur}
    hide={false}
    className="test" searchText="search text" />)

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders hidden', () => {
  const wrapper = shallow(<Search
    {...mockStyles}
    placeholder={placeholder}
    onChange={onChange}
    onKeyDown={onKeyDown}
    onFocus={onFocus}
    onBlur={onBlur}
    hide={true}
    className="test" />)

  expect(toJson(wrapper)).toMatchSnapshot()
})

it('forwards ref', () => {
  const ref = React.createRef()
  const wrapper = mount(<Search
    {...mockStyles}
    placeholder={placeholder}
    onChange={onChange}
    onKeyDown={onKeyDown}
    onFocus={onFocus}
    onBlur={onBlur}
    hide={false}
    ref={ref} />)

  expect(ref.current).toBeDefined()
})

it('should keyDown', () => {
  const wrapper = shallow(<Search
    {...mockStyles}
    placeholder={placeholder}
    onChange={onChange}
    onKeyDown={onKeyDown}
    onFocus={onFocus}
    onBlur={onBlur}
    hide={false} />)

  wrapper.find('input').simulate('keyDown')

  expect(onKeyDown).toBeCalled()
})

it('should change', () => {
  const wrapper = shallow(<Search
    {...mockStyles}
    placeholder={placeholder}
    onChange={onChange}
    onKeyDown={onKeyDown}
    onFocus={onFocus}
    onBlur={onBlur}
    hide={false} />)

  wrapper.find('input').simulate('change')

  expect(onChange).toBeCalled()
})
