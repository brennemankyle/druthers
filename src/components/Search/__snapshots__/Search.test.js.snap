// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<input
  className="test search"
  disabled={false}
  onChange={[MockFunction]}
  onKeyDown={[MockFunction]}
  placeholder="Placeholder"
  type="text"
/>
`;

exports[`renders disabled 1`] = `
<input
  className="test search"
  disabled={true}
  onChange={[MockFunction]}
  onKeyDown={[MockFunction]}
  placeholder="Placeholder"
  type="text"
/>
`;

exports[`renders hidden 1`] = `
<input
  className="test search"
  disabled={false}
  onChange={[MockFunction]}
  onKeyDown={[MockFunction]}
  placeholder="Placeholder"
  type="text"
/>
`;

exports[`renders searchText 1`] = `
<input
  className="test search"
  disabled={false}
  onChange={[MockFunction]}
  onKeyDown={[MockFunction]}
  placeholder="Placeholder"
  type="text"
  value="search text"
/>
`;
